flexget/components/notify/notifiers/telegram.py

```diff
@@ -299,6 +312,14 @@ def _send_msgs(self, msg, chat_ids):
             try:
                 logger.debug('sending msg to telegram servers: {}', msg)
                 self._bot.sendMessage(chat_id=chat_id, text=msg, **kwargs)
+            except ChatMigrated as e:
+                logger.debug("Chat migrated to id {}", e.new_chat_id)
+                try:
+                    self._bot.sendMessage(chat_id=e.new_chat_id, text=msg, **kwargs)
+                    self._replace_chat_id(chat_id, e.new_chat_id, session)
```

Comment:
I don't really know anything about telegram, or this plugin, but should replacing the id happen before retrying sending the message?

Reply:
Hello @gazpachoking 
Yes, because when we send the message we get the exception that indicates a new ID and the new ID

Reply:
I just meant should line 318 and 319 be reversed here in the handling of the ID change.

Reply:
The message send in line 318 is the second attempt to send the message. Line 319 is to update the new id in the database if the message was successfully send. I only update the new id if I'm able to send the message, that is reason I need to be in that order, basically:

- Send message
- Message not send because new id
- Send message with new id
- Message send, update new id in the database
- 
Notice that a chat changing the id does not occurs that often, normally when changing some configs in the groups the chat goes from normal group to supergroup, and this changes the id... I don't think it will change again... And now since the new id is already in the database, all will be good

Reply:
Gotcha, sounds good.